DuckDuckGo/AppConfigurationFetch.swift

```diff
@@ -95,6 +94,16 @@ class AppConfigurationFetch {
     static private var shouldRefresh: Bool {
         return Date().timeIntervalSince(Self.lastConfigurationRefreshDate) > Constants.minimumConfigurationRefreshInterval
     }
+
+    enum BackgroundRefreshCompletionStatus {
+        case expired
+        case noData
+        case newData
+
+        var success: Bool {
+            self == .noData || self == .newData
```

Comment:
Yeah that's a fair suggestion, I wrote it this way originally in order to explicitly state the expected cases, to avoid the situation where new cases added in the future won't impact the logic – let's say a new `.error` case is added, the current implementation would still work as intended, whereas only checking for `.expired` would mean `success` returns true even for an `.error`.

All that being said, this enum is so simple that checking for `expired` makes sense. 🙂

Reply:
Ho yes I agree with your point of view 😏
The only dangerous stuff with computed properties like this is when there is no unit test on it, which could involve one day the typical issue your are mentioning ; So maybe just unit test it maybe? 🤷‍♂️